lighthouse-core/gather/gatherers/dobetterweb/doctype.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+/**
+ * Get and return `name`, `publicId`, `systemId` from
+ * `document.doctype`
+ * @return {Object<string>}
+ */
+function getDoctype() {
+  const {name, publicId, systemId} = document.doctype; // eslint-disable-line no-undef
```

Comment:
this will throw if there's no doctype, so I guess have an early return as well?
```js
if (!document.doctype) {
  return null;
}
...
```

Reply:
much better, thx